shacl12-core/index.html

```diff
@@ -5507,6 +5508,185 @@ <h4>sh:property</h4>
 					</p>
 				</section>
 				
+				<section id="MemberShapeConstraintComponent">
+					<h4>sh:memberShape</h4>
+					<p>
+						<code>sh:memberShape</code> requires that each <a>value node</a> be a <a href="#syntax-rule-SHACL-list">SHACL list</a>,
+						and that each member of that list conforms to the given <a>node shape</a>.
+					</p>
+					<p>
+						<span class="component-class">Constraint Component IRI</span>: <code>sh:MemberShapeConstraintComponent</code>
+					</p>
+
+					<div class="parameters">Parameters:</div>
+					<table class="term-table">
+						<tr>
+							<th>Property</th>
+							<th>Summary and Syntax Rules</th>
+						</tr>
+						<tr>
+							<td><code>sh:memberShape</code></td>
+							<td>
+								The shape that all members of the <a href="#syntax-rule-SHACL-list">SHACL list</a> must conform to.
+								<span data-syntax-rule="memberShape-node">The value of <code>sh:memberShape</code> must be a <a>well-formed</a> <a>node shape</a>.</span>
+								<span data-syntax-rule="memberShape-maxCount">A shape has at most one value for <code>sh:memberShape</code>.</span>
```

Comment:
Is there a firm design reason to cap `sh:memberShape` at 1?  I could imagine a use case wanting to reuse multiple `sh:NodeShape`s that are also used outside the current `sh:PropertyShape`.

Reply:
> Is there a firm design reason to cap `sh:memberShape` at 1?

If we enable multiple then we need to define whether we want all (i.e. `sh:and`) or some (i.e. `sh:or`) `?memberShape`s to be satisfied for the validation to pass; and implementations need to support this.

Equivalent behaviour can be achieved by having those writing shapes define a `?memberShape` that uses `sh:or` or `sh:and` within the shape to achieve the same behaviour. Granted, this is slightly more verbose for those writing shapes.

Reply:
All constraint components only allow a single value per constraint parameter. So having two sh:memberShapes would just execute them individually, resulting in AND semantics. For OR semantics, wrap them into sh:or.